### ISSUE TITLE
URN resolution not working for all namespaces in the URNs

### ISSUE THREAD

**Issue by ctschroeder** [CopticScriptorium/cts#149]
This is a recurring issue (see #126). I'm not sure why it keeps re-appearing (or the URN resolution keeps breaking).

The resolvable URNs are listed at the bottom of each document, as [here in AP 19 under "Cite this Document"](https://data.copticscriptorium.org/texts/ap/apophthegmata-patrum-sahidic-019-presbyter/). However, only the first full URN resolves.

When we originally designed this program:

- entering the complete document cts URN (in the metadata field of that name) would return the document
- entering the URN left-to-right up through any namespace will return all documents with those elements in their cts URNs
- entering any part of an existing URN with a wildcard in one of the namespaces would return all documents according to those parameters (so urn:cts:copticLit:shenoute.*.monbya would return all documents beginning with URNs beginning with urn:cts:copticLit:shenoute. and ending with .monbya

Somehow currently only the first one works. Really all three should, but we do promise on every document that at least items one and two will work.

Hopefully the information in the old issue #126 will aid in resolving this quickly.

**Comment by lgessler** [comment]
Just talked to @amir-zeldes about this--so I think what happened is that when we overhauled the UI in #117 a lot of the internals around URN handling were reimplemented. I did my best to copy all the behaviors and ask y'all if what I had so far was correct, but as we found out in #126 there were still some regressions. As for these issues, I think these are all distinct from the issues raised on #126 and so are also regressions introduced by #117. 

I'll get to this in the coming couple weeks.

**Comment by ctschroeder** [comment]
Thank you so much for looking into this. I thought we had tested the urns when you finished work on 126. I apologize if I missed something. 

Let me know what I can do to help. 

Thank you again. 

Sent from my iPhone

**Comment by lgessler** [comment]
OK, should be fixed. @amir-zeldes feel free to test and deploy

**Comment by ctschroeder** [comment]
Great news. I will test it out this weekend. I have a couple big deadlines this week. Thanks @lgessler !!!!!

**Comment by lgessler** [comment]
np! TBC @amir-zeldes will need to deploy this before it will be available on data.copticscriptorium.org

**Comment by ctschroeder** [comment]
Ok is this deployed on a test site currently? Thanks!

**Comment by amir-zeldes** [comment]
OK, it's deployed - let me know if everything looks good

**Comment by lgessler** [comment]
I don't think the update has happened yet--did you restart the app after you pulled @amir-zeldes ?

**Comment by amir-zeldes** [comment]
I restarted apache - is there something else that needs to be restarted?

**Comment by lgessler** [comment]
Think that's it--looks good now

**Comment by amir-zeldes** [comment]
OK, will wait for feedback from @ctschroeder too

**Comment by ctschroeder** [comment]
Hi! This is looking good to me. I texted a dozen corpora by clicking on multiple links down in the "how to cite" section. Everything worked properly. The only thing I noticed that is different from what we intended but not critical is that the redirects to GitHub go to the main folder for that document's corpus in the public Corpora repo, not to the specific TEI or PAULA or relANNIS subfolders in that corpus's folder. (Hope that makes sense.) I think we can close if everyone is ok with that.
